scripts/build/helm-publisher.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+set -e
+
+owner="containers"
+repo="nri-plugins"
+index_file="index.yaml"
+
+git checkout gh-pages
+
+# Fetch browser download URLs for releases
+releases=$(curl -s "https://api.github.com/repos/$owner/$repo/releases" | jq -r '.[].assets[].browser_download_url')
+
+# Check if releases were found
+if [ -n "$releases" ]; then
+    # Loop through the URLs
+    for url in $releases; do
+        # Check if the URL path exists in index.yaml
+        # and if not, update the index.yaml accordingly
+        if ! grep -q "$url" "$index_file"; then
+            wget "$url"
+            base_url=$(dirname "$url")
+            if ! helm repo index . --url "$base_url" --merge "$index_file"; then
+                echo "Failed to update "$index_file" for: $base_url"
+            fi
```

Comment:
but if we happen to have two URLs for charts with the same base from GitHub, do we really have to deal with it here? I feel like it should be avoided earlier, in other words, before publishing the release.

Reply:
I mean when you run `helm repo index` it adds all the charts in the directoty (if I'm not mistaken). I mean when you download `releases/download/v0.1.0/chart-v0.1.0.tgz` it will add that wit the correct baseurl. Then you download `releases/download/v0.2.0/chart-v0.2.0.tgz` but stale `chart-v0.1.0.tgz` is still there and will be re-added with the incorrect `releases/download/v0.2.0/` baseuril, right? Otherwise, why call repo index N times when you could just wget all and run repo index once?

Reply:
No, we don't download the tarballs we just obtain the links and compare them to the index.yaml file.

Reply:
I mean, `helm index repo `in our case is not updating index.yaml based on its knowledge, instead we dictate it to add the URL. So, there is no tarball in the current directory that can be noticed by the `helm index repo `.

Reply:
Hmm, but the url is the baseurl 🧐 AFAIU with that command nothing actually gets added/merged in the repo index (as there are no charts to be added there)

Reply:
From the `helm repo index` help:
```
Read the current directory and generate an index file based on the charts found.
```

Reply:
ah sorry, I was wrong. You are right. I forgot in my last comment that there is wget and your point about clean up before exiting the every iteration of the loop makes sense. I will fix it.